src/openeo_gfmap/features/feature_extractor.py

```diff
@@ -0,0 +1,285 @@
+        lon, lat = np.meshgrid(lon, lat)
+
+        if self.epsg is None:
+            raise Exception(
+                "EPSG code was not defined, cannot extract lat/lon array "
+                "as the CRS is unknown."
+            )
+
+        # If the coordiantes are not in EPSG:4326, we need to reproject them
+        if self.epsg != 4326:
+            # Initializes a pyproj reprojection object
+            transformer = Transformer.from_crs(
+                crs_from=CRS.from_epsg(self.epsg),
+                crs_to=CRS.from_epsg(4326),
+                always_xy=True,
+            )
+            lon, lat = transformer.transform(xx=lon, yy=lat)
+
+        # Create a two channel numpy array of the lat and lons together by stacking
+        latlon = np.stack([lat, lon])
+
+        # Repack in a dataarray
+        return xr.DataArray(
+            latlon,
+            dims=["bands", "y", "x"],
+            coords={
+                "bands": [LAT_HARMONIZED_NAME, LON_HARMONIZED_NAME],
+                "y": inarr.coords["y"],
+                "x": inarr.coords["x"],
+            },
+        )
+
+    def _execute(self, cube: XarrayDataCube, parameters: dict) -> XarrayDataCube:
+        arr = cube.get_array().transpose("bands", "t", "y", "x")
+        arr = self._common_preparations(arr, parameters)
+        arr = self.execute(arr).transpose("bands", "y", "x")
+        return XarrayDataCube(arr)
+
+    @abstractmethod
+    def execute(self, inarr: xr.DataArray) -> xr.DataArray:
+        pass
+
+
+class PointFeatureExtractor(FeatureExtractor):
+    def __init__(self):
+        raise NotImplementedError(
+            "Point based feature extraction on Vector Cubes is not supported yet."
+        )
+
+    def _execute(self, cube: XarrayDataCube, parameters: dict) -> XarrayDataCube:
+        arr = cube.get_array().transpose("bands", "t")
+
+        arr = self._common_preparations(arr, parameters)
+
+        outarr = self.execute(cube.to_array()).transpose("bands", "t")
+        return XarrayDataCube(outarr)
+
+    @abstractmethod
+    def execute(self, inarr: xr.DataArray) -> xr.DataArray:
+        pass
+
+
+def generate_udf_code(feature_extractor_class: FeatureExtractor) -> openeo.UDF:
+    """Generates the udf code by packing imports of this file, the necessary
+    superclass and subclasses as well as the user defined feature extractor
+    class and the apply_datacube function.
+    """
+    import inspect
+
+    # UDF code that will be built here
+    udf_code = ""
+
+    assert issubclass(
+        feature_extractor_class, FeatureExtractor
+    ), "The feature extractor class must be a subclass of FeatureExtractor."
+
+    if issubclass(feature_extractor_class, PatchFeatureExtractor):
+        udf_code += f"{REQUIRED_IMPORTS}\n\n"
+        udf_code += f"{inspect.getsource(FeatureExtractor)}\n\n"
+        udf_code += f"{inspect.getsource(PatchFeatureExtractor)}\n\n"
+        udf_code += f"{inspect.getsource(feature_extractor_class)}\n\n"
```

Comment:
FYI: idea for future improvement: instead of manually  putting the source of FeatureExtractor, PatchFeatureExtractor, feature_extractor_class here, you could also walk the inheritance trail automatically